src/Action/CreateCommentAction.php

```diff
@@ -60,20 +66,35 @@ final class CreateCommentAction extends Controller
      */
     private $mailer;
 
+    /**
+     * @var EventDispatcherInterface|null
+     */
+    private $eventDispatcher;
+
+    // NEXT_MAJOR: Make $eventDispatcher a required dependency
     public function __construct(
         RouterInterface $router,
         BlogInterface $blog,
         PostManagerInterface $postManager,
         CommentManagerInterface $commentManager,
         FormFactoryInterface $formFactory,
-        MailerInterface $mailer
+        MailerInterface $mailer,
+        EventDispatcherInterface $eventDispatcher = null
     ) {
         $this->router = $router;
         $this->blog = $blog;
         $this->postManager = $postManager;
         $this->commentManager = $commentManager;
         $this->formFactory = $formFactory;
         $this->mailer = $mailer;
+        $this->eventDispatcher = $eventDispatcher;
+
+        if (null === $this->eventDispatcher) {
+            @trigger_error(
+                sprintf('Not providing an event dispatcher to %s is deprecated since 3.x', __CLASS__),
```

Comment:
Let's add the package name, and split line length more evenly:

```php
            @trigger_error(sprintf(
                'Not providing an event dispatcher to %s is deprecated since sonata-project/news-bundle 3.x',
                __CLASS__
            ), E_USER_DEPRECATED);
```